test/fs/test_fs_js_api.c

```diff
@@ -469,6 +469,17 @@ void cleanup() {
     remove("closetestfile");
 }
 
+#if WASMFS
```

Comment:
This API is not specific to WASMFS is it?   I see it exists in `libfs.js` too.

Reply:
Oh indeed it does, I had no idea.  But this version of the test is wasm-specific since it uses a backend.  It also looks like the libfs version ignores the "properties" field...

Reply:
I think the goal with the JS API is mostly feature parity with the existing/old FS.   If possible I thin we should avoid extending the JS API.

But yes, if we decide we really do need a new API we should give it a new name.

Would you make making a version of createFile that *is* compatible with the old FS and testing that as part of this PR?  Or as a separate PR?

Reply:
I've made WASMFS createFile compatible with the JS version.  The JS version never used the third parameter, so I'm using it for the backend here.